plotly/Make_a_FigureFactory.md

```diff
@@ -0,0 +1,113 @@
+# Add Your Figure Factory to the Plotly [Python Library](https://plot.ly/python/)
+
+If you have ever wanted to contribute to the Plotly Python Library by adding a new chart type we don't have, now you have the resources to do so. This README will help you get started cloning the plotly.py repo, forking a new branch, creating a new figure factory, and pushing your results to the cloud to get feedback for merging. Just follow all these steps and you'll be ready to go.
+
+## Getting Started:
+1. Clone the `plotly.py` repo locally and then check out the master branch.
+
+```
+$ git clone git@github.com:plotly/plotly.py.git
+$ git fetch origin
+$ git checkout master
+```
+
+2. Checkout a new branch and give it an appropriate name.
+
+```
+$ git checkout -b "my-new-ff"
+```
+
+## Create A figure_factory file
+1. Creating python file and updating `__init__.py`
+
+You are now ready to start writing your code. Begin by moving to the `plotly.figure_factory` directory in the `plotly.py` repo. Notice that there is an `__init__.py` file as well as a bunch of `_figure_factory_chart.py` files in this directory. Each type of unique plotly chart gets its own python file, and the name of each python file is found in the `__init__.py` file.
+
+The inside of the `__init__.py` looks like:
+
+```
+from __future__ import absolute_import
+
+# Require that numpy exists for figure_factory
+import numpy
+
+from plotly.figure_factory._2d_density import create_2d_density
+from plotly.figure_factory._annotated_heatmap import create_annotated_heatmap
+from plotly.figure_factory._candlestick import create_candlestick
+...
+```
+
+If you want to make, for example, a chart called `foo`, then you must create a python file `_foo.py` and then add the following line to the end of `__init__.py`:
+
+```
+from plotly.figure_factory._foo import create_foo
+```
+
+2. Imports
+
+In `_foo.py` write
+
+```
+from __future__ import absolute_import
+```
+
+at line 1. You can add other imports later if you will need them.
+
+3. The main function
+
+It's now time to write the main function `create_foo` that will be called directly by the user. It has the form:
+
+```
+def create_foo(data, height=450, width=600, ...):
```

Comment:
instead of using real attributes (as some people might think they have to add height and width) you could say `attribute1=value, attribute2=value, ...`